### ISSUE TITLE
FEAT: Change integration tests sleep to listen specifically to events

### ISSUE THREAD

**Issue by pappas999** [smartcontractkit/hardhat-starter-kit#59]
As per the comments in https://github.com/smartcontractkit/hardhat-starter-kit/pull/58

Integration tests should listen for the emitted event from the fulfilling call instead of a general wait/sleep before checking the result of the API call or VRF request

**Comment by rogargon** [comment]
Very interested in this feature, especially as a way to locally simulate the presence of an oracle. I've managed to implement it as part of our Chainlink Hackathon contribution.

First, as part of a test. The `OracleRequest` event emitted by the `MockOracle` is monitored and the request is fulfilled automatically, simulating that the oracle provided a response. For instance:

```
   mockOracle.once('OracleRequest',
            (_specId, _sender, requestId, _payment, _cbAddress, _callbackFuncId, expiration, _dataVersion, _data) => {
                console.log('OracleRequest:', requestId, _data)
                mockOracle.fulfillOracleRequest(requestId, numToBytes32(1))
            })
```

In this case, the response is fixed to be `true`, but additional logic can be added to decode the request `_data`. The full test is available from: https://github.com/rogargon/YTVideoNFTknz/blob/main/test/YTVideoNFT.test.js

Additionally, we have developed a Hardhat task that runs in the background and monitors requests to fulfill them. This way, we can interactively test on a local network the frontend, just like the oracle was present through the MockOracle: https://github.com/rogargon/YTVideoNFTknz/blob/main/tasks/mockoracle.js
